networking.md

```diff
@@ -0,0 +1,464 @@
+```
+
+### Egress
+For outgoing isolated network traffic, pods be attached to additional networks using multus. The operators adds the `k8s.v1.cni.cncf.io/networks` annotation that references the list of network-attachment-definitions to which the deployment pods should be attached:
+
+
+```yaml
+annotations:
+  k8s.v1.cni.cncf.io/networks: '[{"name": "internalapi", "namespace": "openstack"},{"name": "storage", "namespace": "openstack"}]'
+```
+
+## Env Pre-requisite
+The NMstate and MetalLB operator need to be deployed and the OCP environment must support Multis CNI
+
+### NMstate operator
+The NMstate operator is performing state-driven network configuration across the cluster nodes with NMState. Further details can be found at:
+
+* [About the Kubernetes NMState Operator](https://docs.openshift.com/container-platform/4.12/networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.html)
+* [The Kubernetes NMState project](https://nmstate.io/kubernetes-nmstate/)
+
+### MetalLB operator
+The MetalLB operator provides a load-balancer implementation for bare metal clusters which brings user traffic into the clusters network. Currently only tested using MetalLB layer 2 mode. Further details can be found at:
+
+* [Upstream Metallb documentation](https://metallb.universe.tf/)
+* [About MetalLB and the MetalLB Operator](https://docs.openshift.com/container-platform/4.12/networking/metallb/about-metallb.html)
+
+### Multus CNI
+Multus CNI enables attaching multiple network interfaces to pods. For each of the networks, a network-attachment-definition defines how the network gets attached to the pod. Further details can be found at:
+
+* [Understanding multiple networks](https://docs.openshift.com/container-platform/4.12/networking/multiple_networks/understanding-multiple-networks.html)
+* [Using the Multus CNI in OpenShift](https://cloud.redhat.com/blog/using-the-multus-cni-in-openshift)
+* [macvlan plugin](https://www.cni.dev/plugins/current/main/macvlan/)
+* [whereabouts IPAM CNI plugin - Extended configuration](https://github.com/k8snetworkplumbingwg/whereabouts/blob/master/doc/extended-configuration.md)
+
+## Steps to configure a controlplane service internal endpoints on an isolated network
+
+### Networks used in this doc
+
+| network name | CIDR | VLAN | Multus CNI | MetalLB IPAddresspool |
+| --- | --- | --- | --- |--- |
+| ctlplane | 192.168.122.0/24 | - | - | - |
+| internalapi | 172.17.0.0/24 | 20 | 172.17.0.30-172.17.0.70 | 172.17.0.80-172.17.0.90 |
+| storage | 172.18.0.0/24 | 21 | 172.18.0.30-172.18.0.70 | - |
+| tenant | 172.19.0.0/24 | 22 | 172.19.0.30-172.19.0.70 | - |
+
+### Connect OCP workers to the isolated networks
+Worker nodes need to be able to connect to the required isolated networks. `NodeNetworkConfigurationPolicy` (`nncp`) resources can be used to apply the desired state using NMstate via the NMstate operator to the worker nodes.
```

Comment:
could drop the "using NMstate" part from "using NMState via the NMstate operator"

Reply:
done